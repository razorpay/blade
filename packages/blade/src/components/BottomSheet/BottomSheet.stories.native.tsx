/* eslint-disable @typescript-eslint/no-explicit-any */
import type { ComponentStory, Meta } from '@storybook/react';
import React from 'react';
import BottomSheet from '@gorhom/bottom-sheet';
import { StyleSheet, Text, View } from 'react-native';
import BaseBox from '~components/Box/BaseBox';

export default {
  title: 'Components/BottomSheet',
} as Meta;

// https://github.com/software-mansion/react-native-reanimated/issues/2994
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
    backgroundColor: 'grey',
  },
  contentContainer: {
    flex: 1,
    alignItems: 'center',
  },
});

const SingleSelectContent = (): React.ReactElement => {
  // ref
  const bottomSheetRef = React.useRef<BottomSheet>(null);

  // variables
  const snapPoints = React.useMemo(() => ['25%', '50%'], []);

  // callbacks
  const handleSheetChanges = React.useCallback((index: number) => {
    console.log('handleSheetChanges', index);
  }, []);

  // renders
  return (
    <View style={styles.container}>
      <BottomSheet
        ref={bottomSheetRef}
        index={1}
        snapPoints={snapPoints}
        onChange={handleSheetChanges}
      >
        <View style={styles.contentContainer}>
          <Text>Awesome ðŸŽ‰</Text>
        </View>
      </BottomSheet>
    </View>
  );
};

const BottomSheetTemplate: ComponentStory<any> = () => {
  return (
    <BaseBox>
      <SingleSelectContent />
    </BaseBox>
  );
};

export const Default = BottomSheetTemplate.bind({});
// Need to do this because of storybook's weird naming convention, More details here: https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy#single-story-hoisting
Default.storyName = 'Default';
