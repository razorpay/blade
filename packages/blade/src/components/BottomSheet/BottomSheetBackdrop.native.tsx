import React from 'react';
import type { BottomSheetBackdropProps } from '@gorhom/bottom-sheet';
import { BottomSheetBackdrop as GorhomBottomSheetBackdrop } from '@gorhom/bottom-sheet';
import { useTheme } from '~components/BladeProvider';

const BottomSheetBackdrop = (
  props: BottomSheetBackdropProps & { zIndex: number },
): React.ReactElement => {
  const { theme } = useTheme();

  return (
    <GorhomBottomSheetBackdrop
      {...props}
      appearsOnIndex={0}
      disappearsOnIndex={-1}
      pressBehavior="close"
      opacity={1}
      style={[
        props.style,
        { backgroundColor: theme.colors.overlay.background, zIndex: props.zIndex },
      ]}
    />
  );
};

export { BottomSheetBackdrop };
