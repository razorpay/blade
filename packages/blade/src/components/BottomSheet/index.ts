export {
  BottomSheet as BottomSheet_PRE_RELEASE,
  BottomSheetBody,
  BottomSheetFooter,
  BottomSheetHeader,
} from './BottomSheet';

export type { BottomSheetProps } from './BottomSheet';
