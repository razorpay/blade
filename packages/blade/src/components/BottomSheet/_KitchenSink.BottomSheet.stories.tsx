import { composeStories } from '@storybook/react';
import * as bottomSheet from './BottomSheet.stories';
import { Box } from '~components/Box';
import { Heading } from '~components/Typography';

const allStories = Object.values(composeStories(bottomSheet));

export const BottomSheet = (): JSX.Element => {
  return (
    <Box display="flex" flexDirection="column" gap="spacing.4">
      {allStories.map((Story) => {
        return (
          <>
            <Heading>{Story.storyName}</Heading>
            <Story />
          </>
        );
      })}
    </Box>
  );
};

export default {
  title: 'Components/KitchenSink/BottomSheet',
  component: BottomSheet,
  parameters: {
    // enable Chromatic's snapshotting only for kitchensink
    chromatic: { disableSnapshot: false },
    options: { showPanel: false },
  },
};
