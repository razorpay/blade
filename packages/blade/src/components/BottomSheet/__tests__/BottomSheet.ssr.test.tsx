/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/ban-ts-comment */
import { BottomSheet, BottomSheetBody, BottomSheetFooter, BottomSheetHeader } from '../BottomSheet';
import renderWithSSR from '~utils/testing/renderWithSSR.web';
import { Text } from '~components/Typography';
import { Badge } from '~components/Badge';
import { Counter } from '~components/Counter';
import { Button } from '~components/Button';

beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
afterAll(() => jest.restoreAllMocks());

describe('<BottomSheet />', () => {
  // Skipping this test as it is fails in SSR with error: ReferenceError: navigator is not defined
  // coming from @floating-ui/react, which is used in BottomSheet component
  // https://github.com/floating-ui/floating-ui/blob/94ba5b7c7069c272c967e4eef6ba09a793880855/packages/utils/react/src/index.ts#L109
  it.skip('should render a BottomSheet ssr', () => {
    const { container } = renderWithSSR(
      <BottomSheet isOpen={true}>
        <BottomSheetHeader
          title="Address Details"
          subtitle="Saving addresses will improve your checkout experience"
          trailing={<Badge variant="positive">Action Needed</Badge>}
          titleSuffix={<Counter variant="positive" value={2} />}
        />
        <BottomSheetBody>
          <Text>BottomSheet body</Text>
        </BottomSheetBody>
        <BottomSheetFooter>
          <Button isFullWidth variant="secondary">
            Remove address
          </Button>
        </BottomSheetFooter>
      </BottomSheet>,
    );

    expect(container).toMatchSnapshot();
  });
});
