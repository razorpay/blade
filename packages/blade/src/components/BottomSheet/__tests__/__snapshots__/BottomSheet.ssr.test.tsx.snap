// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheet /> should render a BottomSheet ssr 1`] = `"<div id="root"><div data-testid="bottomsheet-backdrop" opacity="1" data-blade-component="base-box" class="BaseBox-bmPWx BottomSheetBackdropweb__StyledBottomSheetBackdrop-sc-198k9f6-0 jkMvvn iDnIUr"></div><div data-blade-component="bottom-sheet" data-testid="bottomsheet-surface" aria-modal="true" role="dialog" style="opacity:1;pointer-events:all;height:0;bottom:0;top:auto;z-index:101" class="BottomSheetweb__BottomSheetSurface-qlloie-0 blXGyU"><div data-blade-component="base-box" class="BaseBox-bmPWx gWSczs"><div data-blade-component="BottomSheetGrabHandle" class="BottomSheetGrabHandleweb__BottomSheetGrabHandle-mo12ak-0 iHfLFm"></div><div data-blade-component="bottom-sheet-header" class="BaseBox-bmPWx bONxzW"><div data-blade-component="base-box" class="BaseBox-bmPWx"><div data-blade-component="base-box" class="BaseBox-bmPWx djABsd"><div data-blade-component="base-box" class="BaseBox-bmPWx hJkrmV"><div data-blade-component="base-box" class="BaseBox-bmPWx hXjMxV"><div data-blade-component="base-box" class="BaseBox-bmPWx jxfNRL"><div data-blade-component="base-box" class="BaseBox-bmPWx kDqSNs"><p class="StyledBaseText-dVBfTO ibEuPD" data-blade-component="text">Address Details</p><div data-blade-component="base-box" class="BaseBox-bmPWx jOqpMP"><div data-blade-component="box" class="BaseBox-bmPWx kgBOXG"><div data-blade-component="counter" class="BaseBox-bmPWx ePenKG"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledCounterweb__StyledCounter-sc-1itt9w6-0 igYNjO iHlsLG"><div data-blade-component="base-box" class="BaseBox-bmPWx iUxudu"><p class="StyledBaseText-dVBfTO ivRXAq" data-blade-component="text">2</p></div></div></div></div></div></div><p class="StyledBaseText-dVBfTO kAzdjS" data-blade-component="text">Saving addresses will improve your checkout experience</p></div></div><div data-blade-component="box" class="BaseBox-bmPWx kgBOXG"><button type="button" aria-label="Close" data-blade-component="icon-button" class="StyledIconButtonweb__StyledButton-sc-1f4cg7d-0 jCUyST"><svg aria-hidden="true" data-blade-component="icon" height="20px" viewBox="0 0 24 24" width="20px" fill="none" class="Svgweb__StyledSvg-vcmjs8-0"><path d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z" fill="currentColor" data-blade-component="svg-path"></path></svg></button></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx gJhLSN"><div data-blade-component="box" class="BaseBox-bmPWx"></div></div></div><div data-blade-component="divider" role="separator" class="BaseBox-bmPWx Divider__StyledDivider-sc-8k3avj-0 bWVnPN WrXok"></div></div></div><div data-blade-component="bottom-sheet-body" data-testid="bottomsheet-body" style="-webkit-tap-highlight-color:revert;-webkit-touch-callout:revert;-webkit-user-select:auto;overscroll-behavior:contain;-webkit-overflow-scrolling:touch;user-select:auto;overflow:auto;touch-action:none" class="BaseBox-bmPWx eaEeoi"><div data-blade-component="base-box" class="BaseBox-bmPWx iaqkon"><p class="StyledBaseText-dVBfTO fAQkza" data-blade-component="text">BottomSheet body</p></div></div><div data-blade-component="bottom-sheet-footer" class="BaseBox-bmPWx hZLBAO"><div data-blade-component="divider" role="separator" class="BaseBox-bmPWx Divider__StyledDivider-sc-8k3avj-0 bWVnPN WrXok"></div><div data-blade-component="base-box" class="BaseBox-bmPWx dJEYFx"><button type="button" data-blade-component="button" role="button" class="StyledBaseButtonweb__StyledBaseButton-sc-26bt38-0 dJxnSi"><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedButtonContentweb__AnimatedButtonContent-sc-1fkx0t6-0  jPBClF"><div data-blade-component="base-box" class="BaseBox-bmPWx BaseButton__ButtonContent-zf1huq-0 LeezU esItiH"><div class="StyledBaseText-dVBfTO RwjIC" data-blade-component="base-text">Remove address</div></div></div></button></div></div></div></div></div>"`;

exports[`<BottomSheet /> should render a BottomSheet ssr 2`] = `
.c3.c3.c3.c3.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c5.c5.c5.c5.c5 {
  overflow: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c6.c6.c6.c6.c6 {
  padding-right: 16px;
  padding-left: 16px;
  margin-bottom: 16px;
  margin-top: 16px;
  touch-action: none;
}

.c7.c7.c7.c7.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8.c8.c8.c8.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding-right: 16px;
  margin-right: auto;
}

.c9.c9.c9.c9.c9 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c10.c10.c10.c10.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c12.c12.c12.c12.c12 {
  margin-left: 8px;
}

.c13.c13.c13.c13.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 28px;
}

.c14.c14.c14.c14.c14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c15.c15.c15.c15.c15 {
  min-height: 20px;
  background-color: hsla(150,100%,32%,0.09);
}

.c17.c17.c17.c17.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 8px;
  padding-left: 8px;
}

.c25.c25.c25.c25.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c26.c26.c26.c26.c26 {
  border-bottom-color: hsla(211,20%,52%,0.18);
  border-bottom-style: solid;
}

.c28.c28.c28.c28.c28 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c29.c29.c29.c29.c29 {
  overflow: auto;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-right: 16px;
  padding-left: 16px;
}

.c31.c31.c31.c31.c31 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  z-index: 2;
  margin-top: auto;
  width: 100%;
  background-color: hsla(0,0%,100%,1);
  touch-action: none;
}

.c32.c32.c32.c32.c32 {
  padding: 16px;
}

.c35.c35.c35.c35.c35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 1;
}

.c20.c20.c20.c20.c20 {
  margin-right: 16px;
}

.c21.c21.c21.c21.c21 {
  text-align: left;
  height: 20px;
  background-color: hsla(150,100%,32%,0.09);
}

.c0.c0.c0.c0.c0 {
  position: fixed;
  z-index: 100;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  background-color: hsla(0,0%,0%,0.56);
  opacity: 1;
}

.c24.c24.c24.c24.c24 {
  border: none;
  cursor: pointer;
  padding: 0;
  border-radius: 2px;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: hsla(211,22%,56%,1);
  -webkit-transition-property: color,box-shadow;
  transition-property: color,box-shadow;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
}

.c24.c24.c24.c24.c24:hover {
  color: hsla(211,26%,34%,1);
}

.c24.c24.c24.c24.c24:focus-visible {
  outline: 4px solid hsla(227,100%,59%,0.18);
  outline-offset: 1px;
  -webkit-transition-property: outline-width;
  transition-property: outline-width;
  -webkit-transition-duration: 70ms;
  transition-duration: 70ms;
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  color: hsla(211,26%,34%,1);
}

.c24.c24.c24.c24.c24:active {
  color: hsla(211,26%,34%,1);
}

.c27.c27.c27.c27.c27 {
  border-width: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c11.c11.c11.c11.c11 {
  color: hsla(212,39%,16%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 1rem;
  font-weight: 600;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
  margin-top: 1px;
}

.c18.c18.c18.c18.c18 {
  color: hsla(150,100%,27%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.75rem;
  font-weight: 500;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.125rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: center;
  margin: 0;
  padding: 0;
  overflow: hidden;
  display: -webkit-box;
  line-clamp: 1;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c19.c19.c19.c19.c19 {
  color: hsla(211,22%,56%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.75rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.125rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c30.c30.c30.c30.c30 {
  color: hsla(212,39%,16%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c37.c37.c37.c37.c37 {
  color: hsla(227,100%,59%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 500;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: center;
  margin: 0;
  padding: 0;
}

.c23.c23.c23.c23.c23 {
  color: hsla(150,100%,27%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.75rem;
  font-weight: 500;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.125rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
  overflow: hidden;
  display: -webkit-box;
  line-clamp: 1;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c1.c1.c1.c1.c1 {
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,0,1);
  transition-timing-function: cubic-bezier(0,0,0,1);
  pointer-events: all;
  -webkit-transition-property: opacity;
  transition-property: opacity;
}

.c4.c4.c4.c4.c4 {
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 12px;
  margin-bottom: 4px;
  touch-action: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 100;
}

.c4.c4.c4.c4.c4:after {
  margin: auto;
  content: '';
  width: 56px;
  height: 4px;
  background-color: hsla(211,20%,52%,0.12);
  border-radius: 16px;
}

.c2.c2.c2.c2.c2 {
  background: hsla(0,0%,100%,1);
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
  border-color: hsla(211,20%,52%,0.18);
  box-shadow: 0px -24px 48px -12px hsla(217,56%,17%,0.18);
  opacity: 0;
  pointer-events: none;
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-timing-function: cubic-bezier(.15,0,.24,.97);
  transition-timing-function: cubic-bezier(.15,0,.24,.97);
  will-change: transform,opacity,height;
  -webkit-transition-property: -webkit-transform,opacity,height;
  -webkit-transition-property: transform,opacity,height;
  transition-property: transform,opacity,height;
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  top: 768px;
  background-color: hsla(0,0%,100%,1);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  touch-action: none;
  overflow: hidden;
}

.c22.c22.c22.c22.c22 {
  background-color: hsla(150,100%,32%,0.09);
  border-radius: 9999px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c16.c16.c16.c16.c16 {
  background-color: hsla(150,100%,32%,0.09);
  border-radius: 9999px;
  max-width: 120px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c33.c33.c33.c33.c33 {
  min-height: 36px;
  width: 100%;
  cursor: pointer;
  background-color: hsla(0,0%,100%,0);
  border-color: hsla(227,100%,59%,1);
  border-width: 1px;
  border-radius: 4px;
  border-style: solid;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 20px;
  padding-right: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition-property: background-color,border-color,box-shadow;
  transition-property: background-color,border-color,box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  position: relative;
}

.c33.c33.c33.c33.c33:hover {
  background-color: hsla(227,100%,59%,0.09);
  border-color: hsla(227,100%,59%,1);
}

.c33.c33.c33.c33.c33:active {
  background-color: hsla(227,100%,59%,0.09);
  border-color: hsla(227,100%,59%,1);
}

.c33.c33.c33.c33.c33:focus-visible {
  background-color: hsla(227,100%,59%,0.09);
  border-color: hsla(227,100%,59%,1);
  outline: 1px solid hsla(227,100%,59%,0.09);
  box-shadow: 0px 0px 0px 4px hsla(227,100%,59%,0.18);
}

.c33.c33.c33.c33.c33 * {
  -webkit-transition-property: color,fill,opacity;
  transition-property: color,fill,opacity;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
}

.c34.c34.c34.c34.c34 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition-duration: cubic-bezier(0.3,0,0.2,1);
  transition-duration: cubic-bezier(0.3,0,0.2,1);
  -webkit-transition-timing-function: 150px;
  transition-timing-function: 150px;
}

.c36.c36.c36.c36.c36 {
  opacity: 1;
}

@media screen and (min-width:768px) {
  .c6.c6.c6.c6.c6 {
    padding-right: 20px;
    padding-left: 20px;
    margin-bottom: 20px;
    margin-top: 20px;
  }
}

@media screen and (min-width:320px) {
  .c26.c26.c26.c26.c26 {
    border-bottom-style: solid;
  }
}

@media screen and (min-width:480px) {
  .c26.c26.c26.c26.c26 {
    border-bottom-style: solid;
  }
}

@media screen and (min-width:768px) {
  .c26.c26.c26.c26.c26 {
    border-bottom-style: solid;
  }
}

@media screen and (min-width:1024px) {
  .c26.c26.c26.c26.c26 {
    border-bottom-style: solid;
  }
}

@media screen and (min-width:1200px) {
  .c26.c26.c26.c26.c26 {
    border-bottom-style: solid;
  }
}

@media screen and (min-width:768px) {
  .c32.c32.c32.c32.c32 {
    padding: 20px;
  }
}

<div
  id="root"
>
  <div
    class="c0 c1"
    data-blade-component="base-box"
    data-testid="bottomsheet-backdrop"
    opacity="1"
  />
  <div
    aria-modal="true"
    class="c2"
    data-blade-component="bottom-sheet"
    data-testid="bottomsheet-surface"
    role="dialog"
    style="opacity: 1; pointer-events: all; height: 0px; bottom: 0px; z-index: 100;"
  >
    <div
      class="c3"
      data-blade-component="base-box"
    >
      <div
        class="c4"
        data-blade-component="BottomSheetGrabHandle"
      />
      <div
        class="c5"
        data-blade-component="bottom-sheet-header"
      >
        <div
          class=""
          data-blade-component="base-box"
        >
          <div
            class="c6"
            data-blade-component="base-box"
          >
            <div
              class="c7"
              data-blade-component="base-box"
            >
              <div
                class="c8"
                data-blade-component="base-box"
              >
                <div
                  class="c9"
                  data-blade-component="base-box"
                >
                  <div
                    class="c10"
                    data-blade-component="base-box"
                  >
                    <p
                      class="c11"
                      data-blade-component="text"
                    >
                      Address Details
                    </p>
                    <div
                      class="c12"
                      data-blade-component="base-box"
                    >
                      <div
                        class="c13"
                        data-blade-component="box"
                      >
                        <div
                          class="c14"
                          data-blade-component="counter"
                        >
                          <div
                            class="c15 c16"
                            data-blade-component="base-box"
                          >
                            <div
                              class="c17"
                              data-blade-component="base-box"
                            >
                              <p
                                class="c18"
                                data-blade-component="text"
                              >
                                2
                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <p
                    class="c19"
                    data-blade-component="text"
                  >
                    Saving addresses will improve your checkout experience
                  </p>
                </div>
              </div>
              <div
                class="c20"
                data-blade-component="base-box"
              >
                <div
                  class="c13"
                  data-blade-component="box"
                >
                  <div
                    class="c14"
                    data-blade-component="badge"
                  >
                    <div
                      class="c21 c22"
                      data-blade-component="base-box"
                    >
                      <div
                        class="c17"
                        data-blade-component="base-box"
                      >
                        <p
                          class="c23"
                          data-blade-component="text"
                        >
                          Action Needed
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="c13"
                data-blade-component="box"
              >
                <button
                  aria-label="Close"
                  class="c24"
                  data-blade-component="icon-button"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="Svgweb__StyledSvg-vcmjs8-0"
                    data-blade-component="icon"
                    fill="none"
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                  >
                    <path
                      d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
                      data-blade-component="svg-path"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="c25"
              data-blade-component="base-box"
            >
              <div
                class=""
                data-blade-component="box"
              />
            </div>
          </div>
          <div
            class="c26 c27"
            data-blade-component="divider"
            role="separator"
          />
        </div>
      </div>
      <div
        class="c28"
        data-blade-component="bottom-sheet-body"
        data-testid="bottomsheet-body"
        style="-webkit-tap-highlight-color:revert;-webkit-touch-callout:revert;-webkit-user-select:auto;overscroll-behavior:contain;-webkit-overflow-scrolling:touch;user-select:auto;overflow:auto;touch-action:none"
      >
        <div
          class="c29"
          data-blade-component="base-box"
        >
          <p
            class="c30"
            data-blade-component="text"
          >
            BottomSheet body
          </p>
        </div>
      </div>
      <div
        class="c31"
        data-blade-component="bottom-sheet-footer"
      >
        <div
          class="c26 c27"
          data-blade-component="divider"
          role="separator"
        />
        <div
          class="c32"
          data-blade-component="base-box"
        >
          <button
            class="c33"
            data-blade-component="button"
            role="button"
            type="button"
          >
            <div
              class=" c34"
              data-blade-component="base-box"
            >
              <div
                class="c35 c36"
                data-blade-component="base-box"
              >
                <div
                  class="c37"
                  data-blade-component="base-text"
                >
                  Remove address
                </div>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
